kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoKVTable.java

```diff
@@ -109,13 +116,20 @@ public MongoKVTable(
         Indexes.descending(KVDoc.TOMBSTONE_TS),
         new IndexOptions().expireAfter(12L, TimeUnit.HOURS)
     );
-    if (ttl != null) {
-      final Duration expireAfter = ttl.plus(Duration.ofHours(12));
-      final long expireAfterSeconds = expireAfter.getSeconds();
+    if (ttlResolver.isPresent()) {
+      // if the default ttl is infinite  we still have to define the ttl index for
+      // the table since the ttlProvider may apply row-level overrides. To approximate
+      // an "infinite" default retention, we just set the default ttl to the maximum value
```

Comment:
I think we should change the approach here to be more like the Tombstones: we would create the index on a new field `expire_ts` (instead of the insertion ts), which could be just be null/empty for documents that don't have a TTL. And instead of having `expireAfter = ttl + 12h` we should have `expireAfter = 12h` (which by the way, it looks like this change dropped the 12h, which was there to give a grace to make sure that stream time advances past the physical time).  This would dramatically reduce the size of the index in situations where many rows do not have a TTL.

From https://www.mongodb.com/docs/manual/core/index-ttl/#expiration-of-data: 

> If the indexed field in a document doesn't contain one or more date values, the document will not expire.
> 
> If a document does not contain the indexed field, the document will not expire.